Wikipedia/Code/WMFAppViewController.m

```diff
@@ -1045,6 +1062,9 @@ - (SavedArticlesFetcher *)savedArticlesFetcher {
             [[SavedArticlesFetcher alloc] initWithDataStore:[[SessionSingleton sharedInstance] dataStore]
 
                                               savedPageList:[self.dataStore savedPageList]];
+        
+        SavedArticlesFetcherProgressManager* savedArticlesProgressManager = [SavedArticlesFetcherProgressManager shared];
+        savedArticlesProgressManager.fetcher = _savedArticlesFetcher;
```

Comment:
it seems like this only gets set once and is always associated with the fetcher. So I think it could make sense to make this just a property of the fetcher itself. (Unless there is something else I am missing which is entirely possible)

Reply:
Context in my earlier comment: https://github.com/wikimedia/wikipedia-ios/pull/2087/files#r176246309